metadata-service/plugin/src/main/java/com/datahub/plugins/loader/IsolatedClassLoaderImpl.java

```diff
@@ -0,0 +1,294 @@
+package com.datahub.plugins.loader;
+
+import com.datahub.plugins.Plugin;
+import com.datahub.plugins.common.IsolatedClassLoader;
+import com.datahub.plugins.common.PluginConfigWithJar;
+import com.datahub.plugins.common.PluginPermissionManager;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.ProtectionDomain;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+
+/**
+ * IsolatedClassLoader to load custom implementation of DataHub Plugins.
+ */
+@Slf4j
+public class IsolatedClassLoaderImpl extends ClassLoader implements IsolatedClassLoader {
```

Comment:
100% needs a test

Reply:
TestPluginFramework is covering all aspects of IsolatedClassLoader, I renamed this test as TestIsolatedClassLoader.